WordPress/Classes/Services/SharingSyncService.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+import CocoaLumberjack
+import WordPressShared
+import WordPressKit
+
+/// SharingService is responsible for wrangling publicize services, publicize
+/// connections, and keyring connections.
+///
+@objc class SharingSyncService: CoreDataService {
+
+    /// Returns the remote to use with the service.
+    ///
+    /// - Parameter blog: The blog to use for the rest api.
+    ///
+    private func remoteForBlog(_ blog: Blog) -> SharingServiceRemote? {
+        guard let api = blog.wordPressComRestApi() else {
+            return nil
+        }
+
+        return SharingServiceRemote(wordPressComRestApi: api)
+    }
+
+    /// Syncs Publicize connections for the specified wpcom blog.
+    ///
+    /// - Parameters:
+    ///     - blog: The `Blog` for which to sync publicize connections
+    ///     - success: An optional success block accepting no parameters.
+    ///     - failure: An optional failure block accepting an `NSError` parameter.
+    ///
+    @objc open func syncPublicizeConnectionsForBlog(_ blog: Blog, success: (() -> Void)?, failure: ((NSError?) -> Void)?) {
+        let blogObjectID = blog.objectID
+        guard let remote = remoteForBlog(blog) else {
+            failure?(SharingSyncServiceError.siteWithNoRemote as NSError)
+            return
+        }
+
+        remote.getPublicizeConnections(blog.dotComID!, success: { remoteConnections in
+
+            // Process the results
+            self.mergePublicizeConnectionsForBlog(blogObjectID, remoteConnections: remoteConnections, onComplete: success)
+        },
+        failure: failure)
+    }
+
+    /// Called when syncing Publicize connections. Merges synced and cached data, removing
+    /// anything that does not exist on the server.  Saves the context.
+    ///
+    /// - Parameters:
+    ///     - blogObjectID: the NSManagedObjectID of a `Blog`
+    ///     - remoteConnections: An array of `RemotePublicizeConnection` objects to merge.
+    ///     - onComplete: An optional callback block to be performed when core data has saved the changes.
+    ///
+    fileprivate func mergePublicizeConnectionsForBlog(_ blogObjectID: NSManagedObjectID, remoteConnections: [RemotePublicizeConnection], onComplete: (() -> Void)?) {
```

Comment:
Does this need to be `fileprivate`? Or would `private` be enough?

```suggestion
    private func mergePublicizeConnectionsForBlog(_ blogObjectID: NSManagedObjectID, remoteConnections: [RemotePublicizeConnection], onComplete: (() -> Void)?) {
```

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20047/commits/94898391776bac032ac63786efe2f6512f2e1f04